lib/messageUtils.ts

```diff
@@ -76,7 +78,29 @@ export function toBinary(string: string): string {
   return btoa(String.fromCharCode(...Array.from(new Uint8Array(codeUnits.buffer))));
 }
 
+export function maybeCreateWelcomePreview(message: any): object {
+  if (message.template === "defaultaboutwelcome") {
+    //Shove the about:welcome message in a spotlight
+    let defaultWelcomeFake = {
+      id: message.id,
+      template: "spotlight",
+      targeting: true,
+      content: message,
+    };
+    // Add the modal property to the spotlight to mimic about:welcome
+    defaultWelcomeFake.content.modal = "tab";
+    // The recipe might have a backdrop, but if not, fall back to the default
+    defaultWelcomeFake.content.backdrop = message.backdrop ||
+    "var(--mr-welcome-background-color) var(--mr-welcome-background-gradient)";
+
+    return defaultWelcomeFake;
+  }
+  // if the message isn't about:welcome, just send it back
+  return message;
+}
```

Comment:
Ideally, we'd have a worthwhile test for this, but I think that to do that, we'd need to set up integration testing with a real Firefox build, and that's kinda out of scope here.